core/arch/arm/mm/sp_mem.c

```diff
@@ -46,6 +47,33 @@ struct sp_mem *sp_mem_new(void)
 	return smem;
 }
 
+bool sp_mem_is_shared(struct sp_mem_map_region *new_reg)
+{
+	struct sp_mem *smem = NULL;
+
+	SLIST_FOREACH(smem, &mem_shares, link) {
+		struct sp_mem_map_region *reg = NULL;
+
+		SLIST_FOREACH(reg, &smem->regions, link) {
+			uint64_t reg_end = reg->page_offset +
```

Comment:
You're computing this before you know if you're ever going to need the result. I guess the compiler can optimize this, but it still looks a bit funny.

Reply:
Done